### PR TITLE
feat: warn if `by` column is not sorted in rolling aggregations (as opposed to raising), add warn_if_unsorted argument

### LINKED ISSUE #11225: `rolling_mean` doesn't respect a sorted date column when applied over a group
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl

from datetime import datetime, timedelta

start = datetime(2001, 1, 1)

df_temporal = pl.DataFrame({
    "date": [start + timedelta(days=k) for k in range(25)], 
    "group": ["A"] *13 + ["B"] * 12
}).with_row_count()

df_temporal = df_temporal.set_sorted(pl.col("date"))

df_temporal.with_columns(
    rolling_row_mean=pl.col("row_nr").rolling_mean(
        window_size="2d", by="date", closed="left"
    ).over("group")
)
```

### Log output

_No response_

### Issue description

Hi,
the provided code works for version `0.19.2`, but fails for `0.19.3`.

### Expected behavior

The code should obviously work and return a dataframe.

### Installed versions

<details>

```
--------Version info---------
Polars:              0.19.2
Index type:          UInt32
Platform:            Linux-5.13.0-51-generic-x86_64-with-glibc2.31
Python:              3.10.12 (main, Jun  7 2023, 12:45:35) [GCC 9.4.0]

----Optional dependencies----
adbc_driver_sqlite:  <not installed>
cloudpickle:         <not installed>
connectorx:          <not installed>
deltalake:           <not installed>
fsspec:              <not installed>
matplotlib:          <not installed>
numpy:               1.24.4
pandas:              2.0.3
pyarrow:             <not installed>
pydantic:            <not installed>
sqlalchemy:          <not installed>
xlsx2csv:            <not installed>
xlsxwriter:          <not installed>```

</details>

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #11225

Even if #11134 were addressed, then the warning would still show, because the sortedness flags don't propagate during `over`

### CODE REVIEW

File: py-polars/polars/expr/expr.py [hunk 1]
```diff
@@ -6469,6 +6487,7 @@ def rolling_std(
         by: str | None = None,
         closed: ClosedInterval = "left",
         ddof: int = 1,
+        warn_if_unsorted: bool = True,
```
Comment (ritchie46) [inline py-polars/polars/expr/expr.py:6490]: I think we should flag this argument as experimental? I think the goal is to remove this argument and ensure sortedness is preserved.

Comment (MarcoGorelli) [reply py-polars/polars/expr/expr.py:6490]: Sure, done

Even if it's done on the user's behalf, I think @orlp wanted to give a performance warning if sorting is done on the user's behalf https://github.com/pola-rs/polars/issues/10991#issuecomment-1712042866

**Comment by ritchie46** [comment]
@MarcoGorelli I recalll you said sortedness wasn't preserved in a window function? Is there a root issue I need to fix?

**Comment by MarcoGorelli** [comment]
I started having a look at this:

in 

https://github.com/pola-rs/polars/blob/ce0e54361f646ddba34c572f7de85d0e9bc92541/crates/polars-lazy/src/physical_plan/expressions/apply.rs#L209-L213

then the elements of `acs` have `is_sorted` flags set. E.g. here

https://github.com/pola-rs/polars/blob/ce0e54361f646ddba34c572f7de85d0e9bc92541/crates/polars-lazy/src/physical_plan/expressions/apply.rs#L220-L225

`ac.series` does have a sorted flag

But - when the code gets here:

https://github.com/pola-rs/polars/blob/ce0e54361f646ddba34c572f7de85d0e9bc92541/crates/polars-lazy/src/physical_plan/expressions/apply.rs#L255-L262

then the sorted flag is gone from `s.deep_clone()`

**Comment by cmdlineluser** [comment]
I was curious about this and have been trying to understand what it does, inside `.iter_groups()`:

```rust
[crates/polars-lazy/src/physical_plan/expressions/group_iter.rs:17] &self = AggregationContext {
    state: NotAggregated(
        shape: (25,)
        Series: 'date' [datetime[μs]]
        [
        	2001-01-01 00:00:00
                [...]
        	2001-01-25 00:00:00
        ],
    ),
    groups: Idx(
        GroupsIdx {
            sorted: true,
            first: [
                0,
                13,
            ],
            all: [
                IdxVec: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
                IdxVec: [13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24],
            ],
        },
    ),
    sorted: false,
    update_groups: No,
    original_len: true,
}
```
`df["date"]` is sorted and `GroupsIdx` is sorted - what does the `sorted: false` represent here?

**Comment by ritchie46** [comment]
Yeap, let's merge this for now. Keeping the flags would ideally require us to put them on the arrow arrays. Want to follow up with that later.

**Comment by azmyrajab** [comment]
Thank you!

**Pull request merged**